File: ZI2 2021/lab7/src/Teacher.php
<?php
namespace Wojtek\Lab7;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
/**
 * @ORM\Entity
 *  @ORM\Table(name="teacher")
 */
class Teacher{
    /** 
 * @ORM\Id
 * @ORM\Column(name="id",type="integer")
 * @ORM\GeneratedValue
*/
protected $id;
 /**
 * @ORM\Column(type="string")
 */
protected $firstname;
 /**
 * @ORM\Column(type="string")
 */
protected $lastname;
/**
* @ORM\OneToOne(targetEntity="Classs",inversedBy="teachers", cascade={"persist"})
*/
private $classs;
/**
* @ORM\OneToMany(targetEntity="Subject",mappedBy="teachers", cascade={"persist"})
*/
private $subjects;

public function __construct()
{
    $this->subjects = new ArrayCollection();
}

public function getClasss(){
    return $this->classs;
}
public function setClasss($classs){
    $this->classs = $classs;
}

public function getSubjects(){
    return $this->subjects;
}
public function setSubjects($subjects){
    $this->subjects = $subjects;
}

public function getId(){
    return $this->id;
}
public function setId($id){
    $this->id = $id;
}

public function getLastname(){
    return $this->lastname;
}
public function setLastname($lastname){
    $this->lastname = $lastname;
}

public function getFirstname(){
    return $this->firstname;
}
public function setFirstname($firstname){
    $this->firstname = $firstname;
}
}
?>

File: ZI2 2021/lab5/Teacher.java
import javax.persistence.*;
import java.util.List;

@Entity
public class Teacher {
    @Id
    @GeneratedValue
    private Long id;
    private String firstname;
    private String lastname;
    @OneToOne(cascade = CascadeType.ALL)
    private Class classs;
    @OneToMany(cascade = CascadeType.ALL,mappedBy="teachers")
    private List<Subject> subjects;
    public Long getId() {
        return id;
    }

    public Class getClasss() {
        return classs;
    }

    public void setClasss(Class classs) {
        this.classs = classs;
    }

    public List<Subject> getSubjects() {
        return subjects;
    }

    public void setSubjects(List<Subject> subjects) {
        this.subjects = subjects;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }


}
